examples/custom_events/src/main.rs

```diff
@@ -0,0 +1,185 @@
+///
+///  This is an example showcasing how to build a very simple bot with custom
+/// events  using the matrix-sdk. To try it, you need a rust build setup, then
+/// you can run: `cargo run -p example-custom-events -- <homeserver_url> <user>
+/// <password>`
+///
+/// Use a second client to open a DM to your bot or invite them into some room.
+/// You should see it automatically join. Then post `!ping`  and observe the log
+/// of the bot. You will see that it sends the `Ping` event and upon receiving
+/// it responds with the `Ack` event send to the room. You won't see that in
+/// most regular clients, unless you activate showing of unknown events.
+use std::{env, process::exit};
+
+use matrix_sdk::{
+    config::SyncSettings,
+    room::Room,
+    ruma::{
+        events::{
+            macros::EventContent,
+            room::{
+                member::StrippedRoomMemberEvent,
+                message::{MessageType, OriginalSyncRoomMessageEvent, TextMessageEventContent},
+            },
+            SyncMessageLikeEvent,
+        },
+        OwnedEventId,
+    },
+    Client,
+};
+use serde::{Deserialize, Serialize};
+use tokio::time::{sleep, Duration};
+
+// We use ruma to define our custom events. Just declare the events content
+// by deriving from `EventContent` adn define `ruma_events` for the metadata
+
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[ruma_event(type = "rs.matrix-sdk.example.ping", kind = MessageLike)]
+pub struct PingContent {}
+
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[ruma_event(type = "rs.matrix-sdk.example.ack", kind = MessageLike)]
+pub struct AckContent {
+    // the event ID of the ping.
+    ping_id: OwnedEventId,
+}
+
+/// The final events can be redacted, hence us defining only the specialized
+/// content (and not room_id and event_id and such). Wrapping the content with
+/// `SyncMessageLikeEvent` props them up to become a full event.
+pub type PingEvent = SyncMessageLikeEvent<PingContent>;
```

Comment:
The `EventContent` derive generates type aliases, I think those should be used instead. If `PingContent` was named `PingEventContent` would exist already (but `SyncPingEvent` would be the type alias for this).